TestProject/ClientApp/src/components/Rating.js

```diff
@@ -0,0 +1,82 @@
+﻿import React, { useEffect, useState } from 'react';
+import { NotificationContainer, NotificationManager } from 'react-notifications';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faThumbsUp, faThumbsDown } from '@fortawesome/free-solid-svg-icons'
+
+function Rating({ postId, userProfile, authorId }) {
+    const [rated, setRated] = useState(false);
+    const [rating, setRating] = useState(0);
+
+    useEffect(() => {
+        if (userProfile.id !== authorId) {
+            checkIfRatedByCurrentUser();
+        }
+        getRating();
+    }, [userProfile])
+
+    function handleSetRating(value) {
+        fetch("/rating/set", {
+            method: 'post',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ postId, value })
+        }).then((response) => {
+            if (response.status === 200) {
+                setRated(true);
+            }
+            else {
+                NotificationManager.error("Couldn't set rating");
+            }
+        }).catch(() => {
+            NotificationManager.error("Couldn't set rating");
+        });
+    } 
+
+    function checkIfRatedByCurrentUser() {
+        if (userProfile.loggedIn) {
+            fetch(`rating/check/${postId}`)
+                .then((response) => {
+                    response.json().then((data) => {
+                        setRated(data);
+                    });
+                }).catch(() => {
+                    NotificationManager.error("Couldn't check rating");
+                });
+        }
+    };
+
+    function getRating() {
+        fetch(`rating/get/${postId}`)
```

Comment:
Include rating value into the view model instead making a separate request for it.

Reply:
And pass a handler from Post component in order to update it.